extensions/wikia/AdEngine/js/slot/resolveState.js

```diff
@@ -44,6 +55,58 @@ define('ext.wikia.adEngine.slot.resolveState', [
 		return params;
 	}
 
+	/**
+	 * Check if ad with given adId has been seen in full state
+	 * during last cacheTtl period of time.
+	 *
+	 * @param records array of records in localStorage
+	 * @param adId
+	 * @returns {boolean}
+	 */
+	function wasRecentlySeen(records, adId) {
+		var recentlySeenObj = null;
+
+		records.forEach(function(value) {
+			if (value.adId === adId) {
+				recentlySeenObj = value;
+			}
+		});
+
+		if (recentlySeenObj) {
+			var age = now.getTime() - recentlySeenObj.lastSeenDate;
+
+			if (age > 0 || age < cacheTtl) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	function checkAndUpdateStorage(params) {
+		var age, val, records = cache.get(cacheKey, now) || [];
+
+		if (wasRecentlySeen(records, params.uap)) {
+			log('Full version of uap was seen in last 24h. adId: ' + params.uap, log.levels.debug, logGroup);
+
+			return false;
+		} else {
+			log('Full version of uap was not seen in last 24h. adId: ' + params.uap, log.levels.debug, logGroup);
+
+			val = {
+				adId: uapContext.getUapId() || params.uap,
```

Comment:
if `adId` would be null then we may end up with incorrect object assigned to `recentlySeenObject`. Although it may be an edge case. On the other hand simple check in `if (adId && val.adId === adId)` won't hurt us.